lib/src/int64_native.dart

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: constant_identifier_names
+
+import 'int32.dart';
+import 'int64.dart';
+import 'intx.dart';
+import 'utilities.dart' as u;
+
+/// An immutable 64-bit signed integer, in the range [-2^63, 2^63 - 1].
+/// Arithmetic operations may overflow in order to maintain this range.
+class Int64Impl implements Int64 {
+  final int _i;
+
+  static const Int64Impl MAX_VALUE = Int64Impl(9223372036854775807);
+
+  static const Int64Impl MIN_VALUE = Int64Impl(-9223372036854775808);
+
+  static const Int64Impl ZERO = Int64Impl(0);
+
+  static const Int64Impl ONE = Int64Impl(1);
+
+  static const Int64Impl TWO = Int64Impl(2);
+
+  const Int64Impl([int value = 0]) : _i = value;
+
+  factory Int64Impl.fromInts(int top, int bottom) =>
+      Int64Impl((top << 32) | (bottom & 0xFFFFFFFF));
+
+  factory Int64Impl.fromBytes(List<int> bytes) =>
+      Int64Impl(((bytes[7] & 0xFF) << 56) |
+          ((bytes[6] & 0xFF) << 48) |
+          ((bytes[5] & 0xFF) << 40) |
+          ((bytes[4] & 0xFF) << 32) |
+          ((bytes[3] & 0xFF) << 24) |
+          ((bytes[2] & 0xFF) << 16) |
+          ((bytes[1] & 0xFF) << 8) |
+          (bytes[0] & 0xFF));
+
+  factory Int64Impl.fromBytesBigEndian(List<int> bytes) =>
+      Int64Impl(((bytes[0] & 0xFF) << 56) |
+          ((bytes[1] & 0xFF) << 48) |
+          ((bytes[2] & 0xFF) << 40) |
+          ((bytes[3] & 0xFF) << 32) |
+          ((bytes[4] & 0xFF) << 24) |
+          ((bytes[5] & 0xFF) << 16) |
+          ((bytes[6] & 0xFF) << 8) |
+          (bytes[7] & 0xFF));
+
+  static Int64Impl? parseRadix(String s, int radix, bool throwOnError) {
+    int charIdx = 0;
+    bool negative = false;
+    if (s.startsWith('-')) {
+      negative = true;
+      charIdx++;
+    }
+
+    if (charIdx >= s.length) {
+      if (!throwOnError) return null;
+      throw FormatException('No digits', s, charIdx);
+    }
+
+    int i = 0;
+    for (; charIdx < s.length; charIdx++) {
+      int c = s.codeUnitAt(charIdx);
+      int digit = u.decodeDigit(c);
+      if (digit < radix) {
+        i = (i * radix) + digit;
+      } else {
+        if (!throwOnError) return null;
+        throw FormatException('Not radix digit', s, charIdx);
+      }
+    }
+
+    if (negative) {
+      return Int64Impl(-i);
+    }
+
+    return Int64Impl(i);
+  }
+
+  static int _promote(Object value) {
```

Comment:
Consider marking this as
```
@pragma('vm:prefer-inline')
```

and equivalent dart2wasm annotation (and similar for the emulated one with dart2js annotation)

Reply:
Let's consider it in a separate PR. I want to keep the changes as small as possible to be able to easily test and benchmark, and the current version in `master` branch doesn't have inline pragmas on this function.